Minikube Values

ssalaues opened the pull request:
Minikube values.yml file for light or single machine deployments. It sets up 1 zookeeper and 3 mongos (minimum for a replicaset).

NicolasT commented on charts/minikube.md:
````diff
@@ -0,0 +1,55 @@
+Clone the repo and go into the charts directory
+```shell
+git clone https://github.com/scality/Zenko.git
+cd Zenko/charts
````
This likely won't work right now as-is...

ssalaues replied:
You're right I was thinking too far ahead. I'll add a checkout to ft/k8s-helm for when this gets merged.

NicolasT commented on charts/minikube.md:
````diff
@@ -0,0 +1,55 @@
+
+With your dependencies built, you can run the following shell command to deploy the zenko stack with orbit enabled.
+```shell
+helm install --name zenko --set ingress.enabled=true  --set cloudserver-front.orbit.enabled=true -f single-node-values.yml zenko
````
Why pass `--set` values when you provide a tailored YAML configuration anyway?

ssalaues replied:
I hadn't considered that. I'll set these as the defaults in the single node values.

NicolasT commented on charts/single-node-values.yml:
```diff
@@ -0,0 +1,63 @@
+# Default values for zenko.
```
I wouldn't copy the existing `values.yaml` as you did (or at least partially). This file doesn't contain 'default values'. It's overlayed with the default values from the Chart's `values.yaml`.

NicolasT commented on charts/single-node-values.yml:
```diff
@@ -0,0 +1,63 @@
+    requests:
+      cpu: 500m
+      memory: 256Mi
+  heap: "512M"
```
Then `resources.requests.memory` should be the same, no?

NicolasT commented on charts/single-node-values.yml:
```diff
@@ -0,0 +1,63 @@
+    enabled: false
+
+  nodeExporter:
+    enabled: false
```
All of this should be the case already.

ssalaues commented:
Updated PR with simplified values (only necessary values included). Updated the documentation to reflect changes.

NicolasT commented on charts/single-node-values.yml:
```diff
@@ -0,0 +1,24 @@
+
+prometheus:
+  rbac:
+    create: false
```
Purely because of this, I don't really like this file to be called `single-node-values.yaml`, because indeed it would fail to deploy on a single-node MetalK8s cluster, or most other single-node deployments of a recent K8s version, including GKE.

Could, instead

- *or* Minikube be deployed with RBAC enabled (`--extra-config=apiserver.Authorization.Mode=RBAC`), and things still work,
- *or* this file be renamed to something like `minikube-values.yaml`
- *or* better suggestions :smile:

ssalaues replied:
Yeah I'll play with the RBAC on minikube this way we don't potentially need another values.yml

ssalaues replied:
So I tested with RBAC and seems to work just fine but it add more steps plus makes the current steps seem a bit more cumbersome.

So this would be the modified steps plus the new ones.
```
minikube start --kubernetes-version=v1.9.0 --memory 4096 --extra config=apiserver.Authorization.Mode=RBAC

kubectl create clusterrolebinding add-on-cluster-admin --clusterrole=cluster-admin --serviceaccount=kube-system:tiller

helm init --service-account tiller --wait
```
Which I'm okay with doing but might slightly take away from the "simplicity" perceptions. 

Another option would be to append the instructions in a new section "Installing with RBAC enabled". This way we can take out the prometheus.rbac.false from the yaml file (making it more generically applicable) and simply add it as a --set in the helm install options.

NicolasT commented on charts/single-node-values.yml:
```diff
@@ -0,0 +1,24 @@
+    - ""
+cloudserver-front:
+  orbit:
+    enabled: true
```
Shouldn't this become the default in the `cloudserver-front` Chart, because it's basically broken without this right now? Maybe the option should even be removed until things are fixed again. WDYT @vrancurel

vrancurel replied:
until we have a clear view it is not bad to be explicit

ssalaues commented:
-Updated documentation for optional RBAC setup
-Took out prometheus.rbac.create from single node yml
-Updated the zenko chart notes to show the endpoint in case of the single node

ssalaues commented:
All comments addressed
